step-strikers-app/step-strikers-app/Classes/RPGCharacter.swift

```diff
@@ -0,0 +1,174 @@
+//
+//  RPGCharacter.swift
+//  step-strikers-app
+//
+//  Created by Jalyn Derry on 2/7/23.
+//
+
+class RPGCharacter {
+    var characterName: String
+    var userName: String
+    
+    // Character Stats Variables
+    var maxHealth: Int
+    var currHealth: Int
+    var maxStamina: Int
+    var currStamina: Int
+    
+    // Inventory Variables
+    var weaponsInInventory: [Weapon]
+    var currWeapon: Weapon
+    var armorInInventory: [Armor]
+    var currArmor: Armor
+    
+    // Rolling modifiers
+    var attackModifier = 0
+    var defenseModifier = 0
+    var magicResistanceModifier = 0
+    
+    init(characterName: String, userName: String, health: Int,
+         stamina: Int){
+        self.characterName = characterName
+        self.userName = userName
+        self.maxHealth = health
+        self.currHealth = health
+        self.maxStamina = stamina
+        self.currStamina = stamina
+        
+        // Initialize Starter Inventory
+        self.weaponsInInventory = [fistsWeapon]
```

Comment:
I think it's better if weapon and armor inventory is also passed in. 

I think it's like:
1. Start combat
2. Read data from firebase
3. create all the character, weapon, armor, item, etc, objects we need

So almost every time we call this, the character will already have some stuff in their inventory. Also are we doing item inventory?

Reply:
I didn't put this as a pass-in because they never start with any weapons and armor in their inventory. They just have their fists and the noArmorArmor whenever you initialize a character

Reply:
That's only at the very beginning, like right after they create their character. After they start getting weapon drops, when we create a character at the beginning of combat, they'll already have stuff in their inventory